### ISSUE TITLE
cannot get entire movies for an actor

### ISSUE THREAD

**Issue by ozturkib** [cinemagoer/cinemagoer#445]
#### Issue

I cannot retrieve the full list of movies for an actor. It only retrieves the list of current 14-15 movies. The rest of them can be seen once you click *See all* section on imdb web page. Probably, this is also affected from the change on imdb parser side. 
Example actor on my above code is [Denzel Washington](https://www.imdb.com/name/nm0000243/).

#### Versions
- Python 3.8.5
- Windows 7 64-bit

#### Steps to reproduce the issue

```python
#!/usr/bin/env python

from imdb import Cinemagoer

# create an instance of the Cinemagoer class
ia = Cinemagoer()

code = "0000243" #Denzel Washington

# printing person name
print(ia.get_person(code))
currActor = ia.get_person(code)
ia.update(currActor)


for job in currActor['filmography'].keys():
    print('# Job: ', job)
    for movie in currActor['filmography'][job]:
        print('\t%s %s (role: %s)' % (movie.movieID, movie['title'], movie.currentRole))

```

#### What's the expected result?
I am expecting all movies which can be seen more once you click see all section on imdb page.

#### What's the actual result?

Denzel Washington
# Job:  actor
	17024450 The Equalizer 3 (role: Robert McCall)
	10095582 The Tragedy of Macbeth (role: Macbeth)
	10016180 The Little Things (role: Joe 'Deke' Deacon)
	3766354 The Equalizer 2 (role: Robert McCall)
	6000478 Roman J. Israel, Esq. (role: Roman J. Israel, Esq.)
	2671706 Fences (role: Troy Maxson)
	2404435 The Magnificent Seven (role: Chisolm)
	0455944 The Equalizer (role: Robert McCall)
	1272878 2 Guns (role: Bobby)
	1907668 Flight (role: Whip Whitaker)
	1599348 Safe House (role: Tobin Frost)
	0477080 Unstoppable (role: Frank)
	1037705 The Book of Eli (role: Eli)
	1111422 The Taking of Pelham 123 (role: Walter Garber)
	0427309 The Great Debaters (role: Melvin B. Tolson)
	0765429 American Gangster (role: Frank Lucas)
# Job:  producer
	17024450 The Equalizer 3 (role: producer)
	15507512 The Piano Lesson (role: producer)
	0995854 A Journal for Jordan (role: producer)
	10514222 Ma Rainey's Black Bottom (role: producer)
	3766354 The Equalizer 2 (role: producer)
	6000478 Roman J. Israel, Esq. (role: producer)
	2671706 Fences (role: producer)
	5257770 Tyrese Gibson: Shame (role: producer)
	0455944 The Equalizer (role: producer)
	1599348 Safe House (role: executive producer)
	1037705 The Book of Eli (role: producer)
	0168786 Antwone Fisher (role: producer)
	0265218 Half Past Autumn: The Life and Works of Gordon Parks (role: producer)
	0113254 Hank Aaron: Chasing the Dream (role: executive producer)
# Job:  director
	0995854 A Journal for Jordan (role: Director)
	2671706 Fences (role: Director)
	0413573 Grey's Anatomy (role: Director)
	0427309 The Great Debaters (role: Director)
	0168786 Antwone Fisher (role: Director)

**Comment by K-Mike** [comment]
Hi,

I had the same issue, installed pip install git+https://github.com/cinemagoer/cinemagoer
and it works now.

**Comment by grocovid** [comment]
Having that same issue, i followed K-Mike's suggestion and 
`pip install git+https://github.com/cinemagoer/cinemagoer`

After that,  'filmography' is not a key in currActor anymore . I guess the complete filmography must be accessed differently. Or is it some regression?

**Comment by K-Mike** [comment]
I use 
`def fetch_movies_by_actor(actor_name):
    ia = imdb.IMDb()

    # Search for the actor
    actors = ia.search_person(actor_name)

    if len(actors) == 0:
        print(f"No results found for '{actor_name}'.")
        return

    # Select the first actor from the search results
    actor = actors[0]
    ia.update(actor, info=['filmography'])

    # Get actor's filmography
    movies = actor['filmography'].get('actor', [])
    
#     return movies
    print(f"Movies played by {actor['name']}:")
    for movie in movies:
        print(f"{movie.data['title']}")`

**Comment by grocovid** [comment]
This is a strange one, and indeed K-Mike's code works.
What works:
`ia = imdb.IMDb()
denzel=ia.search_person("Denzel Washington")[0]
ia.update(denzel, info=['filmography'])
print(len(denzel['filmography']['actor']))
`
Yep, you get 62 movies for Denzel Washington. Except that in my projet, I need to pass the imdb person id to my function and this does not work
`ia = imdb.IMDb()
denzel=ia.get_person("0000243")
ia.update(denzel, info=['filmography'])
print(len(denzel['filmography']['actor']))
`
-> KeyError: 'filmography'

And the code originally posted at the top of this thread doesn't work either : in fact the ia.update responds instantaneously in after retrieved using get_person (i.e. probably does not fetch anything from the imdb website), whereas with K-Mike's code it takes a few moments.

**Comment by alberanid** [comment]
that was a bug, thanks for the report!

Fetching the 'main' page will also mark also the 'filmography' data set as fetched, but it's not longer the case since the last IMDb redesign.

Fixed in the 2023.03.19 version.
